src/lib.rs

```diff
@@ -0,0 +1,16 @@
+//! Approximate Lower Bound Arguments (ALBA, <https://eprint.iacr.org/2023/1655.pdf>)
+//!
+//! Alba is a generic protocol to prove succinctly a lower bound of the size of
+//! a, potentially weighted, set. Say we have a set Sp of size |Sp| >= $n_p$,
+//! and a lower bound $n_f$ < $n_p$ of it we want to prove. Alba gives us a
+//! method to generate a proof of knowledge of this bound by finding the
+//! smallest subset of Sp of size $u$ to convince a verifier.
+//! The paper presents several schemes and optimizations. The basic scheme is
+//! enhanced in the "prehashed" version thanks to sorting Sp with a balls and
```

Comment:
The reason why we call this `prehashed` is that we precompute the hashes of the bins in advance. Maybe we could make this clear.